clippy_lints/src/methods.rs

```diff
@@ -1398,6 +1403,45 @@ impl SelfKind {
     }
 }
 
+fn is_astrait(ty: &hir::Ty, self_ty: &hir::Ty, generics: &hir::Generics, name: &[&str]) -> bool {
```

Comment:
I'll rename to `is_as_ref_or_mod_trait`. And it's kinda specific to this lint, what with the self test.